core/store.go

```diff
@@ -328,3 +329,18 @@ func delByPtr(ptr unsafe.Pointer) bool {
 	}
 	return false
 }
+
+func RandomKey() string {
+	if len(keypool) == 0 {
+		return ""
+	}
+	index := rand.Intn(len(keypool))
+	currentIndex := 0
+	for key := range keypool {
```

Comment:
Wouldn't this operation have O(N) complexity? If so there may be a chance of higher latency when the key size is large?

Wouldn't limiting the numbers of iterations to 1024 be enough? 
Do we know how is it implemented in Redis?

Reply:
@JyotinderSingh  & @soumya-codes I  might need some help over here. To implement this RANDOMKEY command. I coundn't able to come up with any solution for O(1) complexity

Reply:
How about taking a very simplistic approach and limiting the number of iterations to within 512/1024.
```
idxRange := len(keypool) / 2
if idxRange > 1024 {
  idxRange = 1024
}

rand.Seed(time.Now().UnixNano())
randomIndex := rand.Intn(idxRange)
```

@JyotinderSingh wdyt?

Reply:
This approach sounds good to me. But it will introduce a very heavy sampling bias towards the start of the keyset.
But I think it is an acceptable trade off at this stage, we fould revisit this if we decide to modify our store data structures in the future.